simplify/src/main/java/org/cf/simplify/Launcher.java

```diff
@@ -146,9 +145,26 @@ public void run(String[] args) throws IOException, UnhandledVirtualException {
         stats.intakeClasses(classNames);
 
         for (String className : classNames) {
+            if (optimizedClasses.contains(className)) {
+                System.out.println("TEST: Already optimized "+className+", skipping");
+                continue;
+            }
             stats.incrementCurrentClassIndex();
             System.out.println("[" + stats.getCurrentClassIndex() + " / " + stats.getTotalClasses() + "] Processing top level class " + className);
-            executeClass(vm, className, stats);
+            try {
+                executeClass(vm, className, stats);
+            } catch (TryAgainException e) {
+                System.out.println("TEST: crashed while optimizing "+className+" writing error.dex");
+                for (String className1 : classManager.getNonFrameworkClassNames()) {
+                    Collection<VirtualMethod> methods = classManager.getVirtualClass(className1).getMethods();
+                    filterMethods(methods, opts.getIncludeFilter(), opts.getExcludeFilter());
+                }
+                Main.errorDexFile = File.createTempFile("error", "dex");
+                classManager.getDexBuilder().writeTo(new FileDataStore(Main.errorDexFile));
+                throw e;
```

Comment:
Rather than throwing an exception here, catching it, and restarting the whole process, I think you could just create a new VM with `vmFactory` using the error dex as your inFile.`